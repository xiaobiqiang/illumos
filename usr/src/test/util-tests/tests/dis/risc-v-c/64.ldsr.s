/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test RV64C-specific loads and stores.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	c.ldsp	s1,(sp)
	c.ldsp	a4,0x40(sp)
	c.ldsp	t3,0x38(sp)
	c.fldsp	fs1,(sp)
	c.fldsp	ft8,0x40(sp)
	c.fldsp	fa5,0x38(sp)
	c.sdsp	t4,(sp)
	c.sdsp	tp,0x20(sp)
	c.fsdsp	ft2,(sp)
	c.fsdsp	fa3,0x20(sp)
.size libdis_test, [.-libdis_test]
