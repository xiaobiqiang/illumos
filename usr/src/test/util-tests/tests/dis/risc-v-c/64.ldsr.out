    libdis_test:      82 64              c.ldsp s1,0x0(sp)
    libdis_test+0x2:  06 67              c.ldsp a4,0x40(sp)
    libdis_test+0x4:  62 7e              c.ldsp t3,0x38(sp)
    libdis_test+0x6:  82 24              c.fldsp fs1,0x0(sp)
    libdis_test+0x8:  06 2e              c.fldsp ft8,0x40(sp)
    libdis_test+0xa:  e2 37              c.fldsp fa5,0x38(sp)
    libdis_test+0xc:  76 e0              c.sdsp t4,0x0(sp)
    libdis_test+0xe:  12 f0              c.sdsp tp,0x20(sp)
    libdis_test+0x10: 0a a0              c.fsdsp ft2,0x0(sp)
    libdis_test+0x12: 36 b0              c.fsdsp fa3,0x20(sp)
